freeipa/src/main/java/com/sequenceiq/freeipa/service/upgrade/UpgradeService.java

```diff
@@ -130,4 +138,22 @@ private List<ImageInfoResponse> getTargetImages(String catalog, Stack stack, Ima
         LOGGER.debug("Found target images: {}", targetImages);
         return targetImages;
     }
+
+    String calculateUpgradeVariant(Stack stack, String accountId) {
+        String variant = stack.getPlatformvariant();
+        boolean migrationEnable = entitlementService.awsVariantMigrationEnable(accountId);
+        if (migrationEnable) {
+            if (AwsConstants.AwsVariant.AWS_VARIANT.variant().value().equals(variant)) {
+                variant = AwsConstants.AwsVariant.AWS_NATIVE_VARIANT.variant().value();
+            }
+        }
+        return variant;
```

Comment:
I think this could be bit more readable:
```
        if (migrationEnable
            && AwsConstants.AwsVariant.AWS_VARIANT.variant().value().equals(currentVariant)) {
                return AwsConstants.AwsVariant.AWS_NATIVE_VARIANT.variant().value();
        } else {
       		return currentVariant;
	}

Reply:
I think not more readable